### DISCUSSION TITLE
Compare Date not working properly

### DISCUSSION THREAD

**Discussion by Adithya-13** [simolus3/drift#1225] [Q&A]
Hello!

I have a function that return a list of task-based on a date, here is the code

```

  Stream<List<TaskWithCategory>> watchAllTaskByDate(DateTime dateTime) {
    final query = (select(tasks)
          ..where((tbl) {
            final date = tbl.deadline;

            return date.year.equals(dateTime.year) &
                date.month.equals(dateTime.month) &
                date.day.equals(dateTime.day);
          })
          ..orderBy(orderTaskByDate())) // order by date
        .join(leftOuterJoinTaskWithCategory()); // join with other table

    return toTaskWithCategory(query);
  }
```

but the where clause will only return true if the hour of deadline `date` up to 7 AM if I set the hour of  `date` below 7 AM like 6 AM, it will return false. 

I don't know what happens, thank you all, sorry for my bad English

**Comment by simolus3** [comment]
__Edit__: See also the updated discussion in https://github.com/simolus3/drift/issues/1933#issuecomment-1189561299 for a better solution.

I think this is a timezone issue, but so far I'm not sure how it can happen. When you write date time values, moor stores the unix timestamp of the value. Moor then uses the regular [date and time functions](https://sqlite.org/lang_datefunc.html) to extract individual fields. These functions should use the local date time of the device so I wouldn't expect a mismatch here. Do you use utc date times when inserting values by any chance?

If you only need to store a date (and don't care about the time), it may make sense to store it as a string and use a [type converter](https://moor.simonbinder.eu/docs/advanced-features/type_converters/) to convert it to a date time with the format you need.

**Reply by Adithya-13** [reply]
I appreciate your answer. It's working, so when I save it to the database, I used `toLocal()` function in my DateTime deadline. and for the parameter function DateTime on `watchAllTaskByDate()` i used `toLocal()` function too. Thank u
